### PR TITLE
feat: add `ws_uri` config to `ape-node`

### PR DESCRIPTION
**Pull request by antazoey**
### What I did

allows users to configure a `ws_uri` in their node config (per network, same as uri).

### How I did it

implement a custom `ws_uri` in 
### Checklist

- [ ] All changes are completed
- [ ] New test cases have been added
- [ ] Documentation has been updated

### CODE REVIEW

**Review by fubuloubu — DISMISSED** [review]

File: docs/userguides/networks.md [hunk 1]
```diff
@@ -367,7 +367,15 @@ To configure network URIs in `node`, you can use the `ape-config.yaml` file:
 node:
   ethereum:
     mainnet:
-      uri: https://foo.node.bar
+      # For `uri`, you can use either HTTP, WS, or IPC values.
```
Comment (fubuloubu) [inline docs/userguides/networks.md:370]: I forgot about IPC! IPC actually supports `eth_subscribe` too (and it's faster/more reliable)

**Review by fubuloubu — APPROVED** [review]

File: src/ape_ethereum/provider.py [hunk 2]
```diff
@@ -1205,9 +1205,11 @@ class EthereumNodeProvider(Web3Provider, ABC):
     def uri(self) -> str:
         if "url" in self.provider_settings:
             raise ConfigError("Unknown provider setting 'url'. Did you mean 'uri'?")
-        elif "uri" in self.provider_settings and _is_http_url(self.provider_settings["uri"]):
-            # Use adhoc, scripted value
-            return self.provider_settings["uri"]
+        elif uri := self.provider_settings.get("uri"):
+            if _is_http_url(uri) or _is_ws_url(uri):
```
Comment (fubuloubu) [inline src/ape_ethereum/provider.py:1209]: Would this work with IPC?

File: docs/userguides/networks.md [hunk 3]
```diff
@@ -365,11 +365,16 @@ To configure network URIs in `node`, you can use the `ape-config.yaml` file:
 
 ```yaml
 node:
+  # When managing or running a node, configure its IPC path directly (optional)
+  ipc_path: path/to/geth.ipc
```
Comment (fubuloubu) [inline docs/userguides/networks.md:369]: Does under here make sense?

Comment (antazoey) [reply docs/userguides/networks.md:369]: Do you mean the config path? It was there already from before, but yeah it's annoying it doesn't align with the others per-network.

We can add both.. Have this one serve as a global for all networks and the other one to be per network...

Comment (antazoey) [reply docs/userguides/networks.md]: 4e8d2f3

File: docs/userguides/networks.md [hunk 4]
```diff
@@ -380,7 +396,7 @@ The following example shows how to do this.
 (note: even though this example uses `ethereum:mainnet`, you can use any of the L2 networks mentioned above, as they all have these config properties).
 
 ```yaml
-ethereum:
+ethereum:
```
Comment (fubuloubu) [inline docs/userguides/networks.md:399]: mistake?

**Review by fubuloubu — DISMISSED** [review]
Really exciting, but when we migrate the subscriptions manager code upstream, we can actually use IPC Providers as well for silverback automatically

**Pull request merged**